web/admin/src/modal.js

```diff
@@ -133,26 +130,14 @@ export class CustomModal extends Component {
       this.props.confirmDelete(task)
     }
 
-    confirmEdit = (task, value) => {
-      var status = true
-      var named = value.trim()
-      if (named) {
-        for (var item of this.props.sessions){
-          if (item.sessionName === named){
-            status = false
-          }
-        }
+    confirmEdit = (task, value, status) => {
         if (status){
-          this.props.confirmEdit(task, named)
+          this.props.confirmEdit(task, value)
           this.setState({color: "#FAFAFA"});
         }
-        else {
-          this.setState({color: "red", reload: true, task});
+        if (status === false) {
```

Comment:
Can `status` be something other than `true` or `false`?  If it is always a Boolean, the if/else should be enough and slightly more readable than 2 `if`s.